src/app/features/tech-record/components/tech-promote/tech-promote.component.ts

```diff
@@ -0,0 +1,49 @@
+import { Component } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { CustomFormControl, CustomFormGroup, FormNodeTypes } from '@forms/services/dynamic-form.types';
+import { StatusCodes, TechRecordModel, VehicleTechRecordModel } from '@models/vehicle-tech-record.model';
+import { Store } from '@ngrx/store';
+import { TechnicalRecordService } from '@services/technical-record/technical-record.service';
+import { updateEditingTechRecord, updateTechRecords } from '@store/technical-records';
+import { TechnicalRecordServiceState } from '@store/technical-records/reducers/technical-record-service.reducer';
+import cloneDeep from 'lodash.clonedeep';
+import { Observable, tap } from 'rxjs';
+
+@Component({
+  selector: 'app-tech-promote',
+  templateUrl: './tech-promote.component.html'
+})
+export class TechPromoteComponent {
+  vehicleTechRecord$: Observable<VehicleTechRecordModel | undefined>;
+
+  techRecord!: TechRecordModel;
```

Comment:
Doesn't feel like this should be forced when it depends on a selector and then an arbitrary find in a list. IS it not safer for it to be optional?

Reply:
Agreed, the forcing is now removed.